packages/extension-starter/README.md

```diff
@@ -62,6 +62,39 @@ the extension as they like.
 > Congratulations! The Sample extension was successfully installed! Please visit https://www.npmjs.com/package/@salesforce/extension-starter for more information on how to use this extension.
 ```
 
+# State Management
+
+By default all extensions are enhanced with state management using the `withApplicationExtensionStore` higher-order component. Under the hood
+the state is provided using [Zustand](https://www.npmjs.com/package/zustand) as a global store for the entire PWA-Kit application. 
+Each Application Extension inserts a "slice" into this global store following the 
+[slicing pattern](https://github.com/pmndrs/zustand/blob/37e1e3f193a5e5dec6fbd0f07514aec59a187e01/docs/guides/slices-pattern.md). 
+This allows you to have data separation from one extension to the other when it's important, but also allows you to access state its 
+associated actions of other extensions when needed. 
+
+An examples of why you might want to access state and action from another extension would be opening a store-locator map modal provided via 
+the store-locator extension from other pages like the storefronts toolbar or the base project.
+
+This is how you would do something like this.
+
+```
+// /base-project/app/components/my-component.jsx
+import {useExtensionsStore} from '@salesforce/pwa-kit-extension-sdk/react'
+
+export MyComponent = () => {
+    // Grab the slice of the extension state for "extension-a"
+    const {toggleMapsModal} = useExtensionsStore(
+        (state) =>
+            state.state['@salesforce/extension-store-locator'] || {}
+    )
```

Comment:
I feel like the IDE typescript autocomplete probably won't be able to work with the return value of `useExtensionsStore`? This is a hard problem to solve...

Reply:
Yes, you are right. I have been thinking about this as well, but haven't taken the time to try tackling it. I think what we are going have to do would be to create hooks for each extension slice that is strongly typed. 

For example, if the storefront is going to have a hook to get the state/actions for the store-locator then we would create a hooks for that and strongly type it. 

I suppose this readme is just showing the basic usage of the global store. 

If there is a way to do it automatically I'll do that, but like you said, I think that is going to be hard.

Reply:
We could ask developers to cast the type... not perfect but works for the moment...

```
    const {toggleMapsModal} = useExtensionsStore(
        (state) =>
            state.state['@salesforce/extension-store-locator'] || {}
    ) as StoreLocatorSlice
```

The thing is, are they are to import the type from the other extensions? I think so, right?

Reply:
We could yes. But would be be introducing a dependency that is otherwise optional at that point?

For example, the storefront extension can and should be able to run on it's own, it should have the ability to open the store locator modal if that extension is installed, but it's not a direct dependency. 

Would we simply be making it a dev dependency then? But I'm not sure that would work either because the suggestion here is to explicitly import the type aka making it a normal dependency. 

Interesting problem ....